### ISSUE TITLE
Suggest functional test not working

### ISSUE THREAD

**Issue by zkourouma** [elastic/elasticsearch#7847]
I'm building a small library to interface with elasticsearch more like Django querysets, and part of the functionality involves suggest. In the test suite I have a fake index with mapping and add a bunch of documents using elasticsearch.py with `refresh=True` The issue is with adding extra parameters to term suggest, specifically `min_word_length`, `min_doc_freq`, and `max_term_freq`. 

here is a general overview of the snippets
```
curl -XPUT "http://localhost:9200/foo" -d '{
    "mappings": {
        "my_doc_type": {
            "properties": {
                "location": {
                    "type": "geo_point"
                },
                "foo_loc": {
                    "type": "geo_point"
                },
                "child": {
                    "type": "nested"
                }
            }
        }
    }
}'
curl -XPOST "http://localhost:9200/foo/_refresh"

# Helper function that takes the index name and document json
def add_document(index, document):
    curl -XPOST "http://localhost:9200/foo/my_doc_type?refresh=true&op_type=create" -d document

# This is what the min_word_length query looks like
query = {
  "query": {
    "match_all": {}
  },
  "suggest": {
    "spelling": {
      "text": "bazba",
      "term": {
        "sort": "score",
        "field": "bar",
        "suggest_mode": "missing",
        "max_edits": 2,
        "prefix_length": 1,
        "min_doc_freq": 0,
        "max_term_freq": 0.01,
        "min_word_length": 5
      }
    }
  }
}
```
This query/setup returns suggestions with length 4

**Comment by clintongormley** [comment]
Hi @zkourouma 

Sorry for the delay in responding.  Just looked at your issue description, and you say "you have an issue" but you don't say what it is.  You provide code to index docs, but then don't show us what you're indexing, and you don't explain what results you're expecting and what you get instead.

Is this still an issue for you? If so, could you provide more info?

**Comment by andrewgross** [comment]
The query specifies `"min_word_length": 5`, but it returns results of length 4.

**Comment by clintongormley** [comment]
OK, here's a simple recreation:

    POST t/t/_bulk
    {"index":{}}
    {"foo":"abc"}
    {"index":{}}
    {"foo":"abcd"}
    {"index":{}}
    {"foo":"abcde"}
    {"index":{}}
    {"foo":"abcdef"}
    {"index":{}}
    {"foo":"abcdefg"}
    
    
    GET _suggest
    {
      "spelling": {
        "text": "abcdef",
        "term": {
          "field": "foo",
          "suggest_mode": "missing",
          "prefix_length": 1,
          "min_doc_freq": 0,
          "max_term_freq": 0.01,
          "max_edits": 2,
          "min_word_length":5
        }
      }
    }

This returns `abcd` which is only 4 characters long, not the minimum specified:

    {
      "_shards": {
        "total": 5,
        "successful": 5,
        "failed": 0
      },
      "spelling": [
        {
          "text": "abcdef",
          "offset": 0,
          "length": 6,
          "options": [
            {
              "text": "abcdefg",
              "score": 0.8333333,
              "freq": 1
            },
            {
              "text": "abcd",
              "score": 0.5,
              "freq": 1
            }
          ]
        }
      ]
    }

**Comment by dakrone** [comment]
I just checked locally and this is fixed in 5.0 with Areek's work on the suggester.
